### PR TITLE
Bandwidth Tester

### LINKED ISSUE #9: Bandwidth Tester (single threaded, TCP)
Measure the maximum bandwidth between two servers.

Write a program in `sockets/bandwidth/` that tries to send as much data as possible between the client and server.  Use TCP and a single thread to start and only send data in one direction (from client to server).

### PR DESCRIPTION
**Pull request by phil-lopreiato**
Closes #9 

Sends large packets (1024 bytes) infinitely to the server specified. Depends on multithreaded server (#13/#5), see [Eric's Fork](https://github.com/atryus/adv-net-samples/)

### CODE REVIEW

File: sockets/client-tcp.c [hunk 1]
```diff
@@ -48,6 +48,7 @@ int main(int argc, char ** argv)
                         break;
                 }
         }
+        message = "gr8 b8 m8, r8 8/8";
```
Comment (twood02) [inline sockets/client-tcp.c]: find a way to remove this edit from your pull request.  Hello World is just fine...

Comment (phil-lopreiato) [inline sockets/client-tcp.c]: Must have come in from bb34e16b4eb248eb95fee02438b976838dbe60fe...

File: threads/server-tcp.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+#define BACKLOG 10     // how many pending connections queue will hold
+#define MAX_CLIENTS 512
+
+pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;
+
+void
+*handle_client(void *arg)
+{
+        pthread_mutex_lock(&lock);
```
Comment (twood02) [inline threads/server-tcp.c:26]: What shared state is your lock protecting?

File: threads/server-tcp.c [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+#include <pthread.h>
+
+/****************************************
+        Author: Tim Wood
```
Comment (twood02) [inline threads/server-tcp.c:13]: Add yourselves to the author list (you can bump me to the bottom)

File: threads/server-tcp.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+                pthread_create(&client[i], NULL, (void *)handle_client, (void *)(intptr_t)clientfd);
+/*
+                bytes_read = read(clientfd, message, sizeof message);
+                if(bytes_read < 0) {
+                        perror("ERROR reading socket");
+                }
+                close(clientfd);
+                printf("Read: %s\n", message);
+*/
```
Comment (twood02) [inline threads/server-tcp.c:130]: remove uglies.

File: threads/server-tcp.c (hunk 2 above)
Comment (phil-lopreiato) [inline threads/server-tcp.c:26]: Hang on, this is from the wrong issue - it bases off of #5. Work is still in progress on that one, I'll take this down and add it again. Sorry

**Pull request closed without merging**